client/lib/rpc.rs

```diff
@@ -0,0 +1,279 @@
+use futures::executor;
+use jsonrpc_lite::{JsonRpc, Params};
+use reqwest::Client;
+use serde::Serialize;
+use serde_json::{json, Map, Value};
+
+use casper_execution_engine::core::engine_state::ExecutableDeployItem;
+use casper_node::{
+    crypto::{asymmetric_key::PublicKey, hash::Digest},
+    rpcs::{
+        account::{PutDeploy, PutDeployParams},
+        chain::{GetBlock, GetBlockParams, GetStateRootHash, GetStateRootHashParams},
+        info::{GetDeploy, GetDeployParams},
+        state::{GetBalance, GetBalanceParams, GetItem, GetItemParams},
+        RPC_API_PATH,
+    },
+    types::{BlockHash, Deploy, DeployHash},
+};
+use casper_types::{bytesrepr::ToBytes, Key, RuntimeArgs, URef, U512};
+
+use crate::{
+    deploy::{DeployExt, DeployParams, ListDeploys, SendDeploy, Transfer},
+    error::{Error, Result},
+};
+
+/// Struct representing a single JSON-RPC call to the casper node.
+#[derive(Debug, Default)]
+pub struct RpcCall {
+    // TODO - If/when https://github.com/AtsukiTak/warp-json-rpc/pull/1 is merged and published,
+    //        change `rpc_id` to a `jsonrpc_lite::Id`.
+    rpc_id: u32,
+    node_address: String,
+    verbose: bool,
+}
+
+/// `RpcCall` encapsulates calls made to the casper node service via JSON-RPC.
+impl RpcCall {
+    /// Creates a new RPC instance.
+    ///
+    /// `rpc_id` is used for RPC-ID as required by the JSON-RPC specification, and is returned by
+    /// the node in the corresponding response.
+    ///
+    /// `node_address` identifies the network address of the target node's HTTP server, e.g.
+    /// `"http://127.0.0.1:7777"`.
+    ///
+    /// When `verbose` is `true`, the request will be printed to `stdout`.
+    pub fn new(rpc_id: u32, node_address: String, verbose: bool) -> Self {
+        Self {
+            rpc_id,
+            node_address,
+            verbose,
+        }
+    }
+
+    /// Gets a `Deploy` from the node.
+    pub fn get_deploy(self, deploy_hash: DeployHash) -> Result<JsonRpc> {
+        let params = GetDeployParams { deploy_hash };
+        GetDeploy::request_with_map_params(self, params)
+    }
+
+    /// Queries the node for an item at `key`, given a `path` and a `state_root_hash`.
+    /// - `state_root_hash` - Hash representing a pointer to the current state from which a value
```

Comment:
Adding a blank line here will make the summary in the generated docs more appropriate.  Otherwise the entire doc comment is included in the summary.

```suggestion
    /// Queries the node for an item at `key`, given a `path` and a `state_root_hash`.
    ///
    /// - `state_root_hash` - Hash representing a pointer to the current state from which a value
```